### ISSUE TITLE
plugin does nothing (WP 6.5.2)

### ISSUE THREAD

**Issue by techsvcslib** [pfefferle/wordpress-share-on-bluesky#5]
I'm afraid I have the same problem as previous users have reported. When I activate the plugin and input the necessary credentials, nothing happens, and nothing gets posted to my bsky.social account.

**Comment by pfefferle** [comment]
@techsvcslib can you check the "Debug Informations" on the settings site? Does it show a DID, an Access Token and a Refresh Token?

Otherwise you may have used the wrong credentials?

**Comment by techsvcslib** [comment]
Hello, these fields are all blank/empty. I have verified that the credentials are correct. One thing I noticed that might be helpful, when I used a different web browser, is that when I activated the plugin (before I even was able to input credentials), it spit out the following. Perhaps there is a conflict with another plugin, or with my site settings:

Activation failed: <!DOCTYPE html> Bluesky &lsaquo; The OW Factor &#8212; WordPress addLoadEvent = function(func){if(typeof jQuery!=='undefined')jQuery(function(){func();});else if(typeof wpOnload!=='function'){wpOnload=func;}else{var oldonload=wpOnload;wpOnload=function(){oldonload();func();}}}; var ajaxurl = '/wp-admin/admin-ajax.php', pagenow = 'settings_page_share-on-bluesky', typenow = '', adminpage = 'settings_page_share-on-bluesky', thousandsSeparator = ',', decimalPoint = '.', isRtl = 0; #wpadminbar #wp-admin-bar-wccp_free_top_button .ab-icon:before { content: "\f160"; color: #02CA02; top: 3px; } #wpadminbar #wp-admin-bar-wccp_free_top_button .ab-icon { transform: rotate(45deg); } @media print { #wpadminbar { display:none; } } img.wp-smiley, img.emoji { display: inline !important; border: none !important; box-shadow: none !important; height: 1em !important; width: 1em !important; margin: 0 0.07em !important; vertical-align: -0.1em !important; background: none !important; padding: 0 !important; } window._wpemojiSettings = {"baseUrl":"https:\/\/s.w.org\/images\/core\/emoji\/15.0.3\/72x72\/","ext":".png","svgUrl":"https:\/\/s.w.org\/images\/core\/emoji\/15.0.3\/svg\/","svgExt":".svg","source":{"concatemoji":"https:\/\/theowfactor.org\/wp-includes\/js\/wp-emoji-release.min.js?ver=6.5.2"}}; /*! This file is auto-generated */ !function(i,n){var o,s,e;function c(e){try{var t={supportTests:e,timestamp:(new Date).valueOf()};sessionStorage.setItem(o,JSON.stringify(t))}catch(e){}}function p(e,t,n){e.clearRect(0,0,e.canvas.width,e.canvas.height),e.fillText(t,0,0);var t=new Uint32Array(e.getImageData(0,0,e.canvas.width,e.canvas.height).data),r=(e.clearRect(0,0,e.canvas.width,e.canvas.height),e.fillText(n,0,0),new Uint32Array(e.getImageData(0,0,e.canvas.width,e.canvas.height).data));return t.every(function(e,t){return e===r[t]})}function u(e,t,n){switch(t){case"flag":return n(e,"\ud83c\udff3\ufe0f\u200d\u26a7\ufe0f","\ud83c\udff3\ufe0f\u200b\u26a7\ufe0f")?!1:!n(e,"\ud83c\uddfa\ud83c\uddf3","\ud83c\uddfa\u200b\ud83c\uddf3")&&!n(e,"\ud83c\udff4\udb40\udc67\udb40\udc62\udb40\udc65\udb40\udc6e\udb40\udc67\udb40\udc7f","\ud83c\udff4\u200b\udb40\udc67\u200b\udb40\udc62\u200b\udb40\udc65\u200b\udb40\udc6e\u200b\udb40\udc67\u200b\udb40\udc7f");case"emoji":return!n(e,"\ud83d\udc26\u200d\u2b1b","\ud83d\udc26\u200b\u2b1b")}return!1}function f(e,t,n){var r="undefined"!=typeof WorkerGlobalScope&&self instanceof WorkerGlobalScope?new OffscreenCanvas(300,150):i.createElement("canvas"),a=r.getContext("2d",{willReadFrequently:!0}),o=(a.textBaseline="top",a.font="600 32px Arial",{});return e.forEach(function(e){o[e]=t(a,e,n)}),o}function t(e){var t=i.createElement("script");t.src=e,t.defer=!0,i.head.appendChild(t)}"undefined"!=typeof Promise&&(o="wpEmojiSettingsSupports",s=["flag","emoji"],n.supports={everything:!0,everythingExceptFlag:!0},e=new Promise(function(e){i.addEventListener("DOMContentLoaded",e,{once:!0})}),new Promise(function(t){var n=function(){try{var e=JSON.parse(sessionStorage.getItem(o));if("object"==typeof e&&"number"==typeof e.timestamp&&(new Date).valueOf()<e.timestamp+604800&&"object"==typeof e.supportTests)return e.supportTests}catch(e){}return null}();if(!n){if("undefined"!=typeof Worker&&"undefined"!=typeof OffscreenCanvas&&"undefined"!=typeof URL&&URL.createObjectURL&&"undefined"!=typeof Blob)try{var e="postMessage("+f.toString()+"("+[JSON.stringify(s),u.toString(),p.toString()].join(",")+"));",r=new Blob([e],{type:"text/javascript"}),a=new Worker(URL.createObjectURL(r),{name:"wpTestEmojiSupports"});return void(a.onmessage=function(e){c(n=e.data),a.terminate(),t(n)})}catch(e){}c(n=f(s,u,p))}t(n)}).then(function(e){for(var t in e)n.supports[t]=e[t],n.supports.everything=n.supports.everything&&n.supports[t],"flag"!==t&&(n.supports.everythingExceptFlag=n.supports.everythingExceptFlag&&n.supports[t]);n.supports.everythingExceptFlag=n.supports.everythingExceptFlag&&!n.supports.flag,n.DOMReady=!1,n.readyCallback=function(){n.DOMReady=!0}}).then(function(){return e}).then(function(){var e;n.supports.everything||(n.readyCallback(),(e=n.source||{}).concatemoji?t(e.concatemoji):e.wpemoji&&e.twemoji&&(t(e.twemoji),t(e.wpemoji)))}))}((window,document),window._wpemojiSettings); var analyst_opt_localize = {"nonce":"827e0afe69"}; /* <![CDATA[ */ var userSettings = {"url":"\/","uid":"1","time":"[1713868596](tel:1713868596)","secure":"1"};/* ]]> */ var _wpColorScheme = {"icons":{"base":"#a7aaad","focus":"#72aee6","current":"#fff"}}; if ( window.history.replaceState ) { window.history.replaceState( null, null, document.getElementById( 'wp-admin-canonical' ).href + window.location.hash ); } document.body.className = document.body.className.replace('no-js','js'); (function() { var request, b = document.body, c = 'className', cs = 'customize-support', rcs = new RegExp('(^|\\s+)(no-)?'+cs+'(\\s+|$)'); request = true; b[c] = b[c].replace( rcs, ' ' ); // The customizer requires postMessage and CORS (if the site is cross domain). b[c] += ( window.postMessage && request ? ' ' : ' no-' ) + cs; }()); Skip to main content Skip to toolbar Dashboard DashboardHomeUpdates 0 Posts PostsAll PostsAdd New PostCategoriesTags Photo Gallery Photo GalleryAll GalleriesAdd newCategoriesGeneral SettingsOur products Copy Protection Copy ProtectionSettingsMore Plugins Media MediaLibraryAdd New Media File Pages PagesAll PagesAdd New PageCategoriesTags Comments 00 Comments in moderation Ivory Search Ivory SearchSearch FormsMenu SearchSettingsContact UsUpgrade&nbsp;&nbsp;&#x27a4; Astra AstraDashboardCustomizeSite BuilderSpectra Appearance AppearanceThemes 0PatternsCustomizeWidgetsMenusTheme File Editor Plugins 0 Plugins 0Installed PluginsAdd New PluginPlugin File Editor Users UsersAll UsersAdd New UserProfile Tools ToolsAvailable ToolsImportExportSite Health 0Export Personal DataErase Personal Data Settings SettingsGeneralWritingReadingDiscussionMediaPermalinksPrivacyBlueskyShare on MastodonUpdraftPlus Backups Ultimate Social Media PLUS WP-Optimize WP-OptimizeDatabaseImagesCacheMinifySettingsHelpPremium Upgrade Statistics StatisticsOverviewVisitsOnlineVisitorsReferrersSearch EnginesCountriesPost TypesTaxonomiesAuthorsBrowsersPlatformsTop Visitors TodayOptimizationSettingsAdd-OnsCollapse menu MenuAbout WordPressAbout WordPressGet InvolvedWordPress.orgDocumentationLearn WordPressSupportFeedbackThe OW FactorVisit SiteProtection00 Comments in moderationOnline: 1Today's Visitors: 1Today's Visits: 1Yesterday's Visitors: 13Yesterday's Visits: 23Unlock Mini Chart!NewPostMediaPageUserPurge cachePage cacheCache size: 144 bytes (2 files)Purge cache for all pagesUpdraftPlusBackup / RestoreMigrate / CloneSettingsAdvanced ToolsExtensionsUpdraftPlus PremiumHowdy, The OW FactorThe OW Factor20243_adminEdit ProfileLog Out       .sfsi_plus_show_mobile_setting_notification a { color: #fff; text-decoration: underline; } form.sfsi_plus_mobileNoticeDismiss { display: inline-block; margin: 5px 0 0; vertical-align: middle; } .sfsi_plus_mobileNoticeDismiss input[type='submit'] { background-color: transparent; border: medium none; color: #fff; margin: 0; padding: 0; cursor: pointer; }  Share on Bluesky Bluesky Domain The domain of your Bluesky instance. (This has to be a valid URL including &quot;http(s)&quot;) Bluesky &quot;Identifier&quot; Your Bluesky identifier. Password Your Bluesky application password. It is needed to get an Access-Token and will not be stored anywhere. Debug Informations DID Access Token Refresh Token     (function(d, s, id) { var js, fjs = d.getElementsByTagName(s)[0]; if (d.getElementById(id)) return; js = d.createElement(s); js.id = id; js.src = "//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.5"; fjs.parentNode.insertBefore(js, fjs); }(document, 'script', 'facebook-jssdk')); Thank you for creating with WordPress. Version 6.5.2 (function ($) { $('.fs-submenu-item.wp-support-forum').parent().attr( { target: '_blank', rel: 'noopener noreferrer' } ); })(jQuery); By opting out, we cannot alert you anymore in case of important security updates. In addition, we won’t get pointers how to further improve the plugin based on your integration with our plugin. Ok, don't opt out Opt out Service unavailable. Please try again later (function ($) { var isOptingOut = false $('#analyst-opt-out-modal').appendTo($('body')) $(document).on('click', '.analyst-opt-out', function() { var pluginId = $(this).attr('analyst-plugin-id') $('#analyst-opt-out-modal') .attr({'analyst-plugin-id': pluginId}) .show() }) $('.opt-out-modal-close').click(function () { $('#analyst-opt-out-modal').hide() }) $('#opt-out-action').click(function () { if (isOptingOut) return var $mask = $('#analyst-disable-opt-out-modal-mask') var $error = $('#analyst-opt-out-error') var pluginId = $('#analyst-opt-out-modal').attr('analyst-plugin-id') $mask.show() $error.hide() var self = this isOptingOut = true $(self).text('Opting out...') $.ajax({ url: ajaxurl, method: 'POST', data: { action: 'analyst_opt_out_' + pluginId, nonce: analyst_opt_localize.nonce }, success: function (data) { $(self).text('Opt out') if (data && !data.success) { $('#analyst-opt-out-modal').hide() return } $error.hide() $('#analyst-opt-out-modal').hide() isOptingOut = false var optInAction = $('').attr({ class: 'analyst-action-opt analyst-opt-in', 'analyst-plugin-id': pluginId, 'analyst-plugin-signed': '1' }) .text('Opt In') $('.analyst-opt-out[analyst-plugin-id="'+ pluginId +'"').replaceWith(optInAction) $('[analyst-plugin-id="' + pluginId + '"').attr('analyst-plugin-opted-in', 0) $mask.hide() }, error: function () { $('#analyst-opt-out-error').show() $(self).text('Opt out') } }).done(function () { $mask.hide() isOptingOut = false }) }) })(jQuery) (function ($) { var isOptingIn = false $('#analyst-opt-in-modal').appendTo($('body')) var makeOptIn = function (pluginId) { if (isOptingIn) return isOptingIn = true $.ajax({ url: ajaxurl, method: 'POST', data: { action: 'analyst_opt_in_' + pluginId, nonce: analyst_opt_localize.nonce }, success: function () { $('#analyst-opt-in-modal').hide() isOptingIn = false var optOutAction = $('').attr({ class: 'analyst-action-opt analyst-opt-out', 'analyst-plugin-id': pluginId, 'analyst-plugin-signed': '1' }) .text('Opt Out') $('.analyst-opt-in[analyst-plugin-id="'+ pluginId +'"').replaceWith(optOutAction) $('[analyst-plugin-id="' + pluginId + '"').attr('analyst-plugin-opted-in', 1) } }) } $(document).on('click', '.analyst-opt-in:not([loading])', function() { var pluginId = $(this).attr('analyst-plugin-id') var isSigned = $(this).attr('analyst-plugin-signed') === '1' if (!isSigned) { $('#analyst-install-modal') .attr('analyst-plugin-id', pluginId) .show() return; } $('#analyst-install-modal').attr({'analyst-plugin-id': pluginId}) $(this).attr('loading', true).text('Opting In...') makeOptIn(pluginId); }) $('.opt-in-modal-close').click(function () { $('#analyst-opt-in-modal').hide() }) })(jQuery) Why do you deactivate? Please let us know, so we can improve it! Thank you I couldn't understand how to make it work The plugin didn't work as expected I found a better plugin It's a temporary deactivation Other Please let us know the reason for de-activation. Thank you! Deactivate Cancel (function ($) { $('.deactivate').click(function (e) { var anchor = $(this).find('[analyst-plugin-id]') var pluginId = anchor.attr('analyst-plugin-id') var isOptedIn = anchor.attr('analyst-plugin-opted-in') === '1' // Do not ask for reason if not opted in if (!isOptedIn) { return } e.preventDefault() $('#analyst-deactivate-modal') .attr({ 'analyst-plugin-id': pluginId, 'analyst-redirect-url': $(this).find('a').attr('href') }) .show() }) $('.analyst-deactivate-modal-close').click(function () { $('#analyst-deactivate-modal').hide() }) $('#analyst-deactivation-reasons input[name="deactivation-reason"]').change(function () { $('.question-answer').empty() var root = $('#analyst-deactivation-reasons input[name="deactivation-reason"]:checked').parents('li') $('#analyst-deactivation-error').hide() if (!root.attr('data-input-type')) return var reasonInput = $('<' + root.attr('data-input-type') + '/>').attr({placeholder: root.attr('data-input-placeholder'), class: 'reason-answer'}) root.find('.question-answer').append(reasonInput) }) $('#analyst-disabled-plugin-action').click(function () { var pluginId = $('#analyst-deactivate-modal').attr('analyst-plugin-id') var pluginDeactivationUrl = $('#analyst-deactivate-modal').attr('analyst-redirect-url') var root = $('#analyst-deactivation-reasons input[name="deactivation-reason"]:checked').parents('li'); var reason = root.find('.question-answer .reason-answer').val(); var question = root.find('.question').attr('data-question').trim() var $errorBlock = $('#analyst-deactivation-error') if (!question) { return $errorBlock.show() } $errorBlock.hide() var data = { action: 'analyst_plugin_deactivate_' + pluginId, question: question, nonce: analyst_opt_localize.nonce } if (reason) { data['reason'] = reason.trim(); } $(this).attr('disabled', true).text('Deactivating...'); $('#analyst-disable-deactivate-modal-mask').show(); $.ajax({ url: ajaxurl, method: 'POST', data: data }).done(function () { window.location.href = pluginDeactivationUrl $('#analyst-disable-deactivate-modal-mask').hide(); }) }) })(jQuery) Stay on the safe side Receive our plugin’s alerts in case of critical security, feature & special deal updates and allow non-sensitive diagnostic tracking. Allow & Continue &gt; You’re granting these permissions: Your profile information (name and email) Your site information (URL, WP version, PHP info, plugins & themes) Plugin notices (updates, announcements, marketing, no spam) Plugin events (activation, deactivation and uninstall) Learn more Powered by Sellcodes.com Skip Service unavailable. Please try again later (function ($) { var installPlugin = function (pluginId) { var $error = $('#analyst-install-error') $error.hide() $.ajax({ url: ajaxurl, method: 'POST', data: { action: 'analyst_install_' + pluginId, nonce: analyst_opt_localize.nonce }, success: function (data) { if (data && !data.success) { //error $('#analyst-install-modal').hide() return } window.location.reload() }, error: function () { $('#analyst-install-modal').hide() } }).done(function () { $('#analyst-disable-install-modal-mask').hide() $('#analyst-install-action') .attr('disabled', false) .text('Allow & Continue >') }) } if ($('#analyst-install-modal').attr('analyst-plugin-id')) { $('#analyst-install-modal').show() } $('.analyst-install-modal-close').click(function () { $('#analyst-install-modal').hide() }) $('#analyst-install-action').click(function () { var pluginId = $('#analyst-install-modal').attr('analyst-plugin-id') $('#analyst-install-action') .attr('disabled', true) .text('Please wait...') $('#analyst-disable-install-modal-mask').show() installPlugin(pluginId) }) $('#analyst-permissions-toggle').click(function () { var isVisible = $('#analyst-permissions-block').toggle().is(':visible') isVisible ? $(this).text('Close section') : $(this).text('Learn more') var poweredBy = $('#analyst-powered-by') isVisible ? poweredBy.show() : poweredBy.hide() }) $('#analyst-install-skip').click(function () { var pluginId = $('#analyst-install-modal').attr('analyst-plugin-id') $.post(ajaxurl, { action: 'analyst_skip_install_' + pluginId, nonce: analyst_opt_localize.nonce }).done(function () { $('#analyst-install-modal').hide() }) }) })(jQuery) /* ADMIN NOTICES */ .is-notice { margin:20px 0; padding:0; overflow:hidden; background:#FFF;} .is-notice br {clear: none;} .is-notice-dismiss { display:block; float:right; color:#999; line-height:1; margin:0 0 0 15px; text-decoration:none; } .is-notice-image { float:left; margin:10px; width:90px; height:90px; background:url(https://theowfactor.org/wp-content/plugins/add-search-to-menu/admin/assets/logo.png) no-repeat center; background-size:cover; } .is-notice-body { padding:15px; background:#fff; } .is-notice-content { margin:0 0 10px; padding:0; } .is-notice-links a.button { margin-right: 10px;text-decoration: none;background: #e7f2f7 !important;color: #30667b !important;box-shadow: none;text-shadow: none;} .is-notice-links a.btn-highlight {background: #0071a1 !important;color: #FFF !important;} Close dialog Session expired Please log in again. The login page will open in a new tab. After logging in you can close it and return to this page. wp.i18n.setLocaleData( { 'text direction\u0004ltr': [ 'ltr' ] } ); var heartbeatSettings = {"nonce":"edf7aca0c6"}; var astra_theme_builder = {"title":"Site Builder","rest_url":"\/wp-json\/astra-addon\/v1\/custom-layouts\/","new_custom_layout_base_url":"https:\/\/theowfactor.org\/wp-admin\/post-new.php?post_type=astra-advanced-hook","astra_pricing_page_url":"https:\/\/wpastra.com\/pricing\/","astra_docs_page_url":"https:\/\/wpastra.com\/docs\/custom-layouts-pro\/","admin_url":"https:\/\/theowfactor.org\/wp-admin\/"}; jQuery(function($) { $('div.ud-phpseclib-notice').on('click', 'button.notice-dismiss', function (event) { event.stopImmediatePropagation(); $.ajax(ajaxurl, { type: 'POST', data: { action: 'updraft_ajax', subaction: 'dismiss_phpseclib_notice', nonce: 'd52e7dbcde', }, error: function(xhr, status, error_code) { alert(error_code+':'+status); } }); }); });  if(typeof wpOnload==='function')wpOnload();

**Comment by pfefferle** [comment]
Can you maybe tell me your PHP version?

**Comment by techsvcslib** [comment]
Sure: 8.1.27

**Comment by fatherpat64** [comment]
I have the same issue using the same version of WP (6.5.2) and PHP 8.2. The plugin does nothing, and nothing propagates under debug information.

**Comment by pfefferle** [comment]
@fatherpat64 I removed the image because it showed a password. If it is a valid one, I would recommend you to change it asap!

**Comment by fatherpat64** [comment]
Thank you.

**Comment by pfefferle** [comment]
OK, the error on activation is fixed in 2.1.0 but I can't reproduce the other error.

Can you please check if you are using the right username/password combination?

The username should be either the email address or the custom domain (if you have set one https://bsky.social/about/blog/4-28-2023-domain-handle-tutorial).

**Comment by techsvcslib** [comment]
Thank you very much, @pfefferle. I upgraded and the error on activation no longer appears. But I still did not see any values populating DID, Access Token, and Refresh Token. Then I realized (this wan't obvious to me somehow, sorry) that I needed to remove the @ symbol in the Bluesky "Identifier" field. Once I did that, the Debug values populated as expected. I think I'm all set now. Thanks again.
